orttraining/orttraining/test/python/orttraining_test_onnxblock.py

```diff
@@ -554,6 +554,83 @@ def mse_loss(prediction, target):
         _ = ort_session.run(ort_output_names, ort_inputs)
 
 
+@pytest.mark.parametrize(
+    "block",
+    [SimpleTrainingBlockWithMSELoss, SimpleTrainingBlockWithCrossEntropyLoss, SimpleTrainingBlockWithBCEWithLogitsLoss],
+)
+def test_sgd_optimizer_composition(block):
+    # Given
+    device = "cuda"
+    batch_size, input_size, hidden_size, output_size = 64, 784, 500, 10
+    pt_model, base_model = _get_models(device, batch_size, input_size, hidden_size, output_size)
+    learning_rate = 0.001
+
+    # When / Then no error occurs
+    simple_block = block()
+    for name, _ in pt_model.named_parameters():
+        simple_block.requires_grad(name)
+
+    with onnxblock.base(base_model):
+        _ = simple_block(base_model.graph.output[0].name)
+
+    optimizer = onnxblock.optim.SGD(learning_rate)
+    with onnxblock.empty_base() as accessor:
+        _ = optimizer(simple_block.parameters())
+        optimizer_model = accessor.model
+        assert optimizer_model
+
+
+# TODO: Add a test for correctness when creation of ortvalues of
+# tensor seq is possible on cuda
+def test_sgd_optimizer_execution():
```

Comment:
I wrote a lot of these tests before we had the python bindings.
Can we leverage the python bindings now that we have them to tell whether the graph can be executed correctly using the training api? Maybe you can move this test to `orttraining_test_python_bindings`.